proxy/src/serverless/mod.rs

```diff
@@ -438,6 +439,15 @@ async fn request_handler(
             &config.region,
         );
 
+        ctx.set_user_agent(
+            request
+                .headers()
+                .get(hyper::header::USER_AGENT)
+                .and_then(|h| h.to_str().ok())
+                .map(|h| &h[..USER_AGENT_MAX_LENGTH])
```

Comment:
will it panic if it is shorter than USER_AGENT_MAX_LENGTH?

Reply:
Good catch! Not only that, but the byte index might be in the middle of a multi-byte character.

Reply:
I dropped the truncation. Hyper will make sure the user-agent is not too long.